includes/traits/llms-trait-award-templates-post-list-table.php

```diff
@@ -80,6 +80,24 @@ public function add_post_actions( $actions, $post ) {
 			)
 		);
 
+		$engagement_type = str_replace( 'llms_', '', $post->post_type );
```

Comment:
I think we can refactor the whole method a little bit.
I realize two things here:
1) `$engagement_type` === (basically) `$this->engagement_type`.
2) this method runs twice on each table, that's why we can't use 1)
I think the best will be to turn the trait's private `post_types` array into a protected `post_type` that each class that use this trait will define in their constructor (as you do with the `engagement_type`), and early bail if the `$post->post_type !== $this->post_type`.
What do you think?

Reply:
`LLMS_Trait_Award_Templates_Post_List_Table->add_post_actions()` is called twice for each row in the post table because `LLMS_Trait_Award_Templates_Post_List_Table->award_template_row_actions()` is called by `LLMS_Admin_Post_Table_Achievements->__construct()` and `LLMS_Admin_Post_Table_Certificates->__construct()`.

In `LLMS_Trait_Award_Templates_Post_List_Table`, if we replace 
```php
	protected function award_template_row_actions() {

		add_filter( 'post_row_actions', array( $this, 'add_post_actions' ), 20, 2 );

	}
```
with
```php
	protected function award_template_row_actions() {

		if ( "llms_{$this->engagement_type}" === llms_filter_input( INPUT_GET, 'post_type', FILTER_SANITIZE_STRING ) ) {
			add_filter( 'post_row_actions', array( $this, 'add_post_actions' ), 20, 2 );
		}

	}
```
we won't need to check the post type in `LLMS_Trait_Award_Templates_Post_List_Table->add_post_actions()`.